assets/js/utils/__tests__/dom.spec.ts

```diff
@@ -360,12 +367,27 @@ describe('DOM Utilities', () => {
 
     it('should add click event listener to the document by default', () => {
       const mockCallback = jest.fn();
-      onLeftClick(mockCallback);
+      cleanup = onLeftClick(mockCallback);
 
       fireEvent.click(document.body);
 
       expect(mockCallback).toHaveBeenCalledTimes(1);
     });
+
+    it('should return a cleanup function that removes the listener', () => {
+      const mockCallback = jest.fn();
+      const element = document.createElement('div');
+      const localCleanup = onLeftClick(mockCallback, element as unknown as Document);
+
+      fireEvent.click(element, { button: 0 });
+
+      // Remove the listener
+      localCleanup();
+
+      fireEvent.click(element, { button: 0 });
+
+      expect(mockCallback).toHaveBeenCalledTimes(1);
```

Comment:
To thoroughly test the behavior, you'll need to check this twice: once before you call the cleanup, and once after (this line).